hazelcast-client/src/test/java/com/hazelcast/client/topic/ClientReliableTopicTest.java

```diff
@@ -297,4 +301,49 @@ public void testGetLocalTopicStats() throws Exception {
 
         topic.getLocalTopicStats();
     }
+
+    @Test
+    public void shouldNotBeTerminated_whenClientIsOffline() {
+        final HazelcastInstance ownerMember = hazelcastFactory.newHazelcastInstance();
+
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setConnectionAttemptLimit(Integer.MAX_VALUE);
+        clientConfig.getConnectionStrategyConfig().setReconnectMode(ClientConnectionStrategyConfig.ReconnectMode.ASYNC);
+        String topicName = "topic";
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
```

Comment:
it is better chances to catch the problem if you initialize the client later as possible.

Reply:
I have started the client before second instance on purpose. This is the make sure that first instance is the owner. 
And test was failing consistently without the fix.

Reply:
ok, it was reconnect, i was thinking about async connect.